configuration/pih/petl/jobs/hiv/viral_load/source.sql

```diff
@@ -0,0 +1,126 @@
+SET sql_safe_updates = 0;
+
+DROP TABLE IF EXISTS temp_hiv_constructs;
+DROP TABLE IF EXISTS temp_viral_load;
+DROP TABLE IF EXISTS temp_vl_sample_taken_date;
+DROP TABLE IF EXISTS temp_days_since_vl;
+
+-- hiv constructs table
+CREATE TEMPORARY TABLE temp_hiv_constructs
+(
+patient_id 				INT,
+encounter_id 			INT
+);
+
+INSERT INTO temp_hiv_constructs (patient_id, encounter_id)
+SELECT person_id, encounter_id FROM obs WHERE voided = 0 AND concept_id = CONCEPT_FROM_MAPPING("PIH", "HIV viral load construct") AND encounter_id IN (SELECT encounter_id FROM encounter
+WHERE encounter_type =(SELECT encounter_type_id FROM encounter_type WHERE name = "Laboratory Results"));
+
+-- specimen cpllection date table
+CREATE TEMPORARY TABLE temp_vl_sample_taken_date
+(
+encounter_id 			INT,
+vl_sample_taken_date	DATE
+);
+
+INSERT INTO temp_vl_sample_taken_date(encounter_id, vl_sample_taken_date)
+SELECT encounter_id, DATE(encounter_datetime) FROM encounter WHERE encounter_id IN (SELECT encounter_id FROM temp_hiv_constructs);
```

Comment:
Do we need a separate temp table for sample taken date, separate from "temp_viral_load"?

Reply:
sample taken date is the encounter date. easier way to bring it in is to have a separate table, other approaches throw sub query return more than one row error.